### DISCUSSION TITLE
Forms requiring xblr/ixbr, sourcing submissions, api.

### DISCUSSION THREAD

**Discussion by gety9** [manusimidt/py-xbrl#80] [General]
As title says wonder which filings require xblr/ixblr (and thus can be parsed with py-xblr).

So far i saw obvious 10q/10k, some s-1, is there a full list of forms somewhere? I tried to search on sec but found only 900 page filler's manual :)

Also what's the best way to automate sourcing of fillings/xblr urls?
As i understand one needs supporting script to look up new submissions, choose ixblr/xblr url if it exists and pass it to py-xblr.

**Comment by gety9** [comment]
Also i see that sec recently (announced July 2, 2021) added api for forms 10-Q, 10-K,8-K, 20-F, 40-F, 6-K, and their variants.

refs
https://www.sec.gov/structureddata/announcement/osd-announcement-070221-api-available
https://www.sec.gov/edgar/sec-api-documentation

**Comment by manusimidt** [comment]
**For what filings does the SEC require XBRL:**
This question can't be answered in general. It depends on the company size, industry, and other factors. The SEC introduced XBRL first in 2005. From 2005 till 2009 XBRL was optional. After 2009 all companies that had more than 10 Mio USD in Assets had to file some form types in XBRL. But there are also many exemptions for other companies. After that, the SEC introduced inline XBRL for financial reports (10-K, 10-Q), first for large accelerated filers (2019), then for accelerated filers (2020) and now (since 2021) the use of iXBRL is required for all filers. 
In summary: Unfortunately, the SEC does not provide a clear list (at least I couldn't find one yet) that states which form type has to contain interactive data and which does not. However, it also seems that this would be way too complex for a simple table since it depends on the company. Theoretically, it would be possible for the filers to upload all submissions in iXBRL (https://www.sec.gov/structureddata/osd-inline-xbrl.html) but in my experience, the only filings where you can rely on XBRL are 10-K, 10-Q, 8-K. 

**How to look up xbrl submissions:**
Yes, it is true that you first have to look up xbrl submissions and either download them and provide `py-xbrl` with the path, or just provide the url to the instance document. 
If you want to parse all xbrl files from **ONE company**, I would recommend the API. https://data.sec.gov/submissions/CIK0000320193.json for example gives you all filings from apple. You also can see which filings where filed in xbrl ("isXBRL" - Flag).
If you want to parse all xbrl files of **MANY companies** I would recommend the monthly RSS Feeds. https://www.sec.gov/Archives/edgar/monthly/xbrlrss-2019-06.xml for example gives you all XBRL filings that where submitted in June 2019.
